### DISCUSSION TITLE
How to useFetch only server side?

### DISCUSSION THREAD

**Discussion by iBobik** [nuxt/nuxt#16080] [Questions]
Hello,
I want to fetch some data from a server with CORS incompatible with a browser, so it needs to be fetched server side.

When I reload page it does SSR, so fetch work. But when I change some code, HMR will do the fetch in a browser and it fails on CORS.

I do it like this event with server option:
```
const { data: map } = await useFetch('https://example.org/api/v1/venues/', {
  params: {
    mode: 'full'
  },
  server: true
})
```

**Comment by adriaandotcom** [comment]
For me, it works better with a `key` set:

```js
const { data: map } = await useFetch("https://example.org/api/v1/venues/", {
  params: {
    mode: "full"
  },
  key: "venues"
})
```

Edit: Nope, it does not work during development.

This might be helpful: https://github.com/nuxt/framework/discussions/5396#discussioncomment-2921848

**Reply by adriaandotcom** [reply]
I ended up using a proxy in the API.

```js
// server/api/venues.js
export default defineEventHandler(async (event) => {
  const { token } = useRuntimeConfig();

  const headers = {
    "Content-Type": "application/json",
    Authorization: `Bearer ${token}`,
  };

  const url = "https://example.org/api/v1/venues/";
  return $fetch(url, { method: "GET", headers });
});
```

I needed a token, so added that as well.
